validator_worker/src/producer.rs

```diff
@@ -2,17 +2,28 @@ use std::error::Error;
 
 use chrono::{TimeZone, Utc};
 
-use primitives::adapter::Adapter;
+use primitives::adapter::{Adapter, AdapterErrorKind};
 use primitives::validator::{Accounting, MessageTypes};
-use primitives::BalancesMap;
+use primitives::{BalancesMap, ChannelId};
 
 use crate::core::events::merge_aggrs;
-use crate::sentry_interface::SentryApi;
-use slog::info;
-
-pub type Result = std::result::Result<(BalancesMap, Option<Accounting>), Box<dyn Error>>;
+use crate::sentry_interface::{PropagationResult, SentryApi};
+
+#[derive(Debug)]
+pub enum TickStatus<AE: AdapterErrorKind> {
+    AccountingSent {
+        channel: ChannelId,
+        balances: BalancesMap,
+        new_accounting: Accounting,
+        accounting_propagation: Vec<PropagationResult<AE>>,
+        event_counts: usize,
+    },
+    AccountingNotSent(BalancesMap),
```

Comment:
1. Since this is `producer` specific we can remove the `Accounting` prefixes
2. `AccountingNotSent` relays another meaning, a better descriptive name would be `NoNewEventAggr` or `ZeroEvAggr`
3. For `AccountingSent` we don't need `BalancesMap`, `Accounting`.  `Channel` & `accounting_propagation`, `event_counts` and `state_root`  would be great

Reply:
We do actually need the `BalancesMap`, we use it in `follower::tick` and the `new_accounting` in `leader::tick`